stdlib/public/core/CTypes.swift

```diff
@@ -86,21 +86,35 @@ public typealias CLongDouble = Double
 // which we don't yet have in Swift.
 #if arch(x86_64) || arch(i386)
 public typealias CLongDouble = Float80
-#endif
-// TODO: Fill in definitions for other OSes.
-#if arch(s390x)
+#elseif arch(s390x)
 // On s390x '-mlong-double-64' option with size of 64-bits makes the
 // Long Double type equivalent to Double type.
 public typealias CLongDouble = Double
+#else
+#error("CLongDouble needs to be defined for this Linux architecture")
 #endif
 #elseif os(Android)
 // On Android, long double is Float128 for AAPCS64, which we don't have yet in
 // Swift (https://github.com/apple/swift/issues/51573); and Double for ARMv7.
 #if arch(arm)
 public typealias CLongDouble = Double
+#else
+#error("CLongDouble needs to be defined for this Android architecture")
```

Comment:
As noted above, it is float128 on AArch64, which otherwise works and this is going to break compilation on, same with linux aarch64 above.

Reply:
I am fine with the BSD errors below, but please don't touch linux and Android, as those are already working for aarch64 despite not supporting float128, or require this to be defined for all platforms.

Reply:
OK, I'm happy with that if everyone else is.  Pushed that change now.

Any opinions about the other-OS branch (which I guess would encompass PS4, Cygwin, Haiku, and WASI)?

Reply:
As I said, "don't... require this to be defined for all platforms," as some platforms like linux aarch64 don't define it and work fine, ie remove the last OS error.

Reply:
Done, thanks!